Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -3876,20 +3890,30 @@ - (void)markVisibleMessagesAsRead
 
     TSThread *thread = self.thread;
     uint64_t lastVisibleTimestamp = self.lastVisibleTimestamp;
+
     [self.editingDatabaseConnection asyncReadWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
         NSMutableArray<id<OWSReadTracking>> *interactions = [NSMutableArray new];
+        
         [[TSDatabaseView unseenDatabaseViewExtension:transaction]
             enumerateRowsInGroup:thread.uniqueId
                       usingBlock:^(
                           NSString *collection, NSString *key, id object, id metadata, NSUInteger index, BOOL *stop) {
 
-                          TSInteraction *interaction = object;
-                          if (interaction.timestampForSorting > lastVisibleTimestamp) {
+                          if (![object conformsToProtocol:@protocol(OWSReadTracking)]) {
```

Comment:
We should avoid doing blind casts.

Reply:
So does that mean that in prod there's something in the `unseenDatabaseViewExtension` that doesn't conform to `OWSReadTracking`?

Reply:
The actual crash was a few lines down:

```
[interactions addObject:possiblyRead];
```

Presumably `possiblyRead` is nil - the root cause of that I don't know.